### ISSUE TITLE
console access

### ISSUE THREAD

**Issue by joshmarinacci** [mozilla-mobile/webxr-ios#23]
When running the app from a local XCode build you can access the Javascript console by attaching desktop Safari to it. However, when running the app from the appstore you cannot access the console. I'm assuming this is due to security reasons. Given that this is a tool for developers we need to find a way to give them access to the developer console.

**Comment by blairmacintyre** [comment]
This is an Apple restriction.  There is no way to give them access to the javascript console or content.  Presumably this is to prevent dastardly dastards from peeking inside apps written using web technology and stealing their secrets.

**Comment by joshmarinacci** [comment]
True. However I'd like to leave this open to investigate workarounds.  ex: let the app connect to a websocket server to report errors.

**Comment by blairmacintyre** [comment]
Yes, I agree (that's why I re-opened this after closing it)

http://www.vorlonjs.io/ might be a reasonable solution to suggest.   It can show the console, and do some things.  We (or others) can write custom plugins (e.g., we could do a Webxr plugin that shares the state of webxr and displays it graphically in the client)

**Comment by cvan** [comment]
Perhaps as part of this issue, or a separate issue ought to be filed for this: Using Sentry/raven.js or Google Analytics for collecting JavaScript errors is a good idea.

**Comment by blairmacintyre** [comment]
Can you give me some URLs to show me how this works?  Good idea

**Comment by cvan** [comment]
sure:

- https://blog.sentry.io/2016/01/04/client-javascript-reporting-window-onerror.html
- https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onerror
- https://dev.opera.com/articles/better-error-handling-with-window-onerror/

**Comment by blairmacintyre** [comment]
I wonder how hard it would be to simply add a console view that you can show/hide, that shows all the console output?  @che1404 should we consider this?  Perhaps in the debug view, there's a console button that just pops up a scrollable text view with all of the console messages?

**Comment by che1404** [comment]
Sorry I'm not sure if it's the iOS console or the javascript console what you mean.
If it's the iOS console, maybe we can use this approach, using CocoaLumberJack (that's the one we are using currently): 
https://egeek.me/2012/10/01/log-to-uitextview-on-ios/

**Comment by blairmacintyre** [comment]
I meant the Javascript console.  So, grab the messages and put up a view.

**Comment by che1404** [comment]
I think it should be doable. I'd listen from a specific event type from the javascript, as we do now for the AR data communications, and then we could add it to a UITextview.
